src/inference/src/dev/iplugin.cpp

```diff
@@ -129,52 +138,197 @@ std::unordered_set<std::string> ov::get_supported_nodes(
+                has_consumers = true;
+                if (supported.count(input.get_node()->get_friendly_name())) {
+                    return false;
+                }
+            }
+        }
+        return has_consumers;
+    };
+
+    auto has_users_supported = [&](const NameSet& supported, const NodePtr& node) -> bool {
+        auto users = node->get_users();
+        for (auto& user : users) {
+            if (supported.count(user->get_friendly_name())) {
+                return true;
+            }
+        }
+        return false;
+    };
+
+    auto has_all_sources_supported =
+        [&get_input_node](const NameSet& supported, const NodePtr& op, bool skip_const = true) -> bool {
+        for (auto& input : op->inputs()) {
+            const auto& node = get_input_node(input);
+            if ((skip_const && ov::op::util::is_constant(node)) || (ov::op::util::is_parameter(node)))
+                continue;
+            if (!supported.count(node->get_friendly_name()))
                 return false;
+        }
+        return true;
+    };
+
+    auto has_unsupported_source =
+        [&get_input_node](const NameSet& supported, const NodePtr& op, bool const_only = false) -> bool {
+        for (auto& input : op->inputs()) {
+            const auto& node = get_input_node(input);
+            if (const_only && !ov::op::util::is_constant(node))
+                continue;
+            if (!supported.count(node->get_friendly_name())) {
+                return true;
             }
         }
-        return (node->output(0).get_target_inputs().size() != 0);
+        return false;
     };
 
-    auto has_unsupported_source = [&supported](const std::shared_ptr<ov::Node>& node) {
-        return !supported.count(node->input_values().begin()->get_node()->get_friendly_name());
+    auto remove_op_from_supported = [&](const NodePtr& node) {
+        auto names = get_names_set(node);
+        for (auto& name : get_names_set(node)) {
+            supported.erase(name);
+        }
     };
 
     // Walk over transformed model for special handing of Parameters/Constants/Results
+    bool start_split = false;
+    unsigned long total_size = 0;
+    unsigned long total_ops_size = 0;
+    std::map<std::string, int> pair_checker;
     for (auto&& op : ops) {
-        // Mark Constants and all fused names as unsupported if they are have no
-        // supported consumers/sources
         if (ov::op::util::is_constant(op)) {
-            if (has_all_consumers_unsupported(op)) {
-                auto names = get_names_set(op);
-                for (auto& name : get_names_set(op)) {
-                    supported.erase(name);
+            const auto const_byte_size = op->get_element_type().size() * shape_size(op->get_shape());
+            total_ops_size += const_byte_size;
+        }
+    }
+    if (memory_control) {
+        for (auto&& op : ops) {
+            if (const auto& assign = std::dynamic_pointer_cast<ov::op::util::VariableExtension>(op)) {
```

Comment:
maybe we can stop checking pairs when start_split is true?

Reply:
Yes, updated.